packages/client-sdk-web/src/internal/pubsub-client.ts

```diff
@@ -144,7 +146,11 @@ export class PubsubClient<
       const prepareCallbackOptions: PrepareSubscribeCallbackOptions = {
         ...options,
         restartedDueToError: false,
-        firstMessage: true,
+        // We call subscribe send on retryable errors. All retry-able errors happen
+        // after the first message. Here we can check if there has been a message that
+        // has already been sent from this stream by checking to see if there is a resumable
+        // sequence number. If there is one, we know that this isn't the first message.
+        firstMessage: !options.subscriptionState.resumeAtTopicSequenceNumber,
```

Comment:
i thought you decided that this was a red herring? I don't understand how this comes into play based on my understanding of what the bug was.

Reply:
this is required because our error callback keys off of this `firstMessage` boolean. Normally if the first message is an error we want to end the stream. But if the stream was previously connected, and subsequently severed, we want to make attempts to reconnect the stream. If the first message is always set to true, and we keep getting errors from `subscribeSend` trying to reconnect, the sdk will terminate the connection and stop trying to reconnect. An alternative approach would be to check for `isRstStream` errors first before checking for first message

https://github.com/momentohq/client-sdk-javascript/blob/b42f50310415c8817d04ebc40045efd35b528f6a/packages/core/src/internal/clients/pubsub/AbstractPubsubClient.ts#L171-L207

Reply:
Thanks for the explanation. Better to make the `firstMessage` semantics explicit like you did. The only issue I can think of is we are now coupling the first message detection to the message sequence number. As long as we don't change the latter, we will be fine.

Can you submit an issue for us to come back to this to explicitly model "first message in the user-facing subscription" vs first message in the stream?

Reply:
if we did what Matt said above and "check for isRstStream" first, then we wouldn't need two state variables for "firstMessage" like you are hinting at, right Michael?

feels like a code smell here that we are maybe overloading the semantics of `firstMessage` to mean two different things, and if we can avoid that by simply re-ordering the `if` blocks, that sounds better to me?

Reply:
That works too.

Reply:
@bruuuuuuuce - I know that @danielamiao wants to get this merged and released so that we have it in our pocket in case the customer needs it. So if it would take too much time for you to test the alternate fix of re-ordering the `if` blocks, feel free to merge this and do a release and then just fast-follow tomorrow morning with the update?

Reply:
I'm going to release it as is and fast follow tomorrow with the swapping the if statements fix, my main concern being that changing the ordering of those will affect both the nodejs sdk and the web sdk. So will just require extra testing